File: foundations-jdbc-scala/src/scala/dev/typr/foundationssc/QueryAnalyzer.scala
package dev.typr.foundationssc

object QueryAnalyzer:
  def analyze(analyzable: Analyzable, conn: Connection): List[QueryAnalysis] =
    import scala.jdk.CollectionConverters.*
    dev.typr.foundations.QueryAnalyzer.analyze(analyzable.analyzable, conn).asScala.toList

  def analyze(op: OperationRead[?], conn: Connection): List[QueryAnalysis] =
    import scala.jdk.CollectionConverters.*
    dev.typr.foundations.QueryAnalyzer.analyze(op.underlying, conn).asScala.toList

  def analyze(template: Template[?, ?], conn: Connection): List[QueryAnalysis] =
    import scala.jdk.CollectionConverters.*
    dev.typr.foundations.QueryAnalyzer.analyze(template.underlying, conn).asScala.toList

  @deprecated("Use the overload that takes Connection instead")
  def analyze(analyzable: Analyzable, conn: java.sql.Connection): List[QueryAnalysis] =
    import scala.jdk.CollectionConverters.*
    dev.typr.foundations.QueryAnalyzer.analyze(analyzable.analyzable, conn).asScala.toList

  @deprecated("Use the overload that takes Connection instead")
  def analyze(op: OperationRead[?], conn: java.sql.Connection): List[QueryAnalysis] =
    import scala.jdk.CollectionConverters.*
    dev.typr.foundations.QueryAnalyzer.analyze(op.underlying, conn).asScala.toList

  @deprecated("Use the overload that takes Connection instead")
  def analyze(template: Template[?, ?], conn: java.sql.Connection): List[QueryAnalysis] =
    import scala.jdk.CollectionConverters.*
    dev.typr.foundations.QueryAnalyzer.analyze(template.underlying, conn).asScala.toList

File: foundations-jdbc-kotlin/src/kotlin/dev/typr/foundationskt/QueryAnalyzer.kt
@file:Suppress("unused")
package dev.typr.foundationskt

object QueryAnalyzer {
    @JvmStatic
    fun analyze(analyzable: Analyzable, conn: Connection): List<QueryAnalysis> =
        dev.typr.foundations.QueryAnalyzer.analyze(analyzable.analyzable, conn.javaConnection)

    @JvmStatic
    fun analyze(op: OperationRead<*>, conn: Connection): List<QueryAnalysis> =
        dev.typr.foundations.QueryAnalyzer.analyze(op.underlying, conn.javaConnection)

    @Deprecated("Use the overload that takes Connection instead", ReplaceWith("analyze(analyzable, conn)"))
    @JvmStatic
    fun analyze(analyzable: Analyzable, conn: java.sql.Connection): List<QueryAnalysis> =
        @Suppress("DEPRECATION")
        dev.typr.foundations.QueryAnalyzer.analyze(analyzable.analyzable, conn)

    @Deprecated("Use the overload that takes Connection instead", ReplaceWith("analyze(op, conn)"))
    @JvmStatic
    fun analyze(op: OperationRead<*>, conn: java.sql.Connection): List<QueryAnalysis> =
        @Suppress("DEPRECATION")
        dev.typr.foundations.QueryAnalyzer.analyze(op.underlying, conn)
}
